### DISCUSSION TITLE
How to hide code input in RST files

### DISCUSSION THREAD

**Discussion by fmaussion** [executablebooks/meta#488] [Q&A] [answered]
Hi, I'm trying to convert my sphinx rst RTD theme to the sphinx book theme. I am stuck with RST then.

My question is: how do I convert the following markdown directive:

```
  ```{code-cell} ipython3
  :tags: [remove_cell]
  
  # Generate some code that we'll use later on in the page
  import numpy as np
  import matplotlib.pyplot as plt
  
  square = np.random.randn(100, 100)
  wide = np.random.randn(100, 1000)
  \```
```

into valid rst syntax? I tried:

```
.. ipython:: python
    :hide_input:
```

But the tag isn't recognized. 

Furthermore, I wondered why my outputs look like this:

![image](https://user-images.githubusercontent.com/10050469/133811090-133aead1-608e-4783-a616-bfb519d52f59.png)

While the code snippets on your doc look so pretty (no ipython prompts :star_struck:) : 

![image](https://user-images.githubusercontent.com/10050469/133813166-3c439663-6541-4325-bcce-fbd8b759b0f0.png)

Is there any way to reach the same while staying in rst?

**Comment by chrisjsewell** [comment, marked as answer]
No these only work with markdown, since the document is first converted to a jupyter notebook (unless you added a custom converter)

**Reply by fmaussion** [reply]
Cheers @chrisjsewell - I understand. I wish mardown-sphinx had existed 5 years ago ;-)

**Reply by akhmerov** [reply]
As a quick note, [jupyter-sphinx](https://jupyter-sphinx.readthedocs.io/en/latest/) works in rst (NB: I'm its maintainer). It is much more bare-bones, but it may fit your use case.

**Reply by fmaussion** [reply]
Thanks @akhmerov ! Indeed it works well! The outputs may not look as pretty as in jupyter-book but it does what I need!

**Reply by fmaussion** [reply]
(I will use it - it is really a nice project)

**Reply by akhmerov** [reply]
It would be great if I could piggy-back on the styles defined by the JB, but that requires some more work to happen in defining a common interface for jupyter-related data (https://github.com/executablebooks/meta/issues/152)

**Comment by mmcky** [comment]
@fmaussion glad `jupyter-sphinx` works for your use case -- it's a great project 👍 

If you ever wanted to migrate your `rst` to `myst:md` this project is super handy and makes migration pretty simple. 

https://github.com/executablebooks/rst-to-myst/
